files/usr/share/jeos-firstboot/raspberrywifi

```diff
@@ -122,23 +120,40 @@ WIRELESS_ESSID='$wlan_network'
 WIRELESS_MODE='Managed'
 EOF
 
-	if [ $wlan_auth_mode = "WPA-PSK" ]; then
-		echo "WIRELESS_WPA_PSK='$wlan_password'" >> $config_file
-	fi
+		if [ $wlan_auth_mode = "WPA-PSK" ]; then
+			echo "WIRELESS_WPA_PSK='$wlan_password'" >> $config_file
+		fi
 
-	if [ $wlan_auth_mode = "WPA-EAP" ]; then
-		echo "WIRELESS_WPA_IDENTITY='$wlan_username'" >> $config_file
-		echo "WIRELESS_WPA_PASSWORD='$wlan_password'" >> $config_file
-		echo "WIRELESS_EAP_AUTH='mschapv2'" >> $config_file
-	fi
+		if [ $wlan_auth_mode = "WPA-EAP" ]; then
+			echo "WIRELESS_WPA_IDENTITY='$wlan_username'" >> $config_file
+			echo "WIRELESS_WPA_PASSWORD='$wlan_password'" >> $config_file
+			echo "WIRELESS_EAP_AUTH='mschapv2'" >> $config_file
+		fi
 
-	run mv -f $config_file /etc/sysconfig/network/ifcfg-$wlan_device
+		run mv -f $config_file /etc/sysconfig/network/ifcfg-$wlan_device
+		d --infobox $"Connecting to wireless network ..." 3 38 || true
+
+		run ifdown $wlan_device &>/dev/null || true
+		# Disabling exit on error to capture exit code
+		set +e
+		ifup $wlan_device &>/dev/null
+		ret_val=$?
+		set -e
+		if [ $ret_val -ne 0 ]; then
+			if dialog --yesno $"Connection failed, do you wish to retry?" 0 0; then
```

Comment:
Is it possible to get an error message from wicked? This error message is pretty useless

Reply:
Sadly no, wicked does not have more then a setup-in-progress message, and wpa_supplicant uses a D-bus api to give status, will be more complicated, but I will check on how to get proper errors later.